script/electron-builder.js

```diff
@@ -0,0 +1,104 @@
+const normalizePackageData = require('normalize-package-data');
+const fs = require("fs/promises");
+const generateMetadata = require('./generate-metadata-for-builder')
+
+// Monkey-patch to not remove things I explicitly didn't say so
+// See: https://github.com/electron-userland/electron-builder/issues/6957
+let transformer = require('app-builder-lib/out/fileTransformer')
+const builder_util_1 = require("builder-util");
+
+transformer.createTransformer = function(srcDir, configuration, extraMetadata, extraTransformer) {
+    const mainPackageJson = path.join(srcDir, "package.json");
+    const isRemovePackageScripts = configuration.removePackageScripts !== false;
+    const isRemovePackageKeywords = configuration.removePackageKeywords !== false;
+    const packageJson = path.sep + "package.json";
+    return file => {
+        if (file === mainPackageJson) {
+            return modifyMainPackageJson(file, extraMetadata, isRemovePackageScripts, isRemovePackageKeywords);
+        }
+        if (extraTransformer != null) {
+            return extraTransformer(file);
+        }
+        else {
+            return null;
+        }
+    };
+}
+async function modifyMainPackageJson(file, extraMetadata, isRemovePackageScripts, isRemovePackageKeywords) {
+    const mainPackageData = JSON.parse(await fs.readFile(file, "utf-8"));
+    if (extraMetadata != null) {
+        builder_util_1.deepAssign(mainPackageData, extraMetadata);
+        return JSON.stringify(mainPackageData, null, 2);
+    }
+    return null;
+}
+/// END Monkey-Patch
+
+const builder = require("electron-builder")
+const Platform = builder.Platform
+
+const generate = require('./lib/generate-metadata.js')
+
+let options = {
+  "appId": "link.mauricioszabo.pulsar",
+  "npmRebuild": false,
+  "publish": null,
+  "extraResources": [
+    {
+      "from": "apm",
+      "to": "app/apm"
+    }
+  ],
+  compression: "normal",
+  "linux": {
+    "icon": "resources/app-icons/atom-community.png",
+    "category": "Development",
+    "synopsis": "A hackable text editor for the 22nd century",
+    "target": [
+      {
+        "target": "appimage",
+        "arch": "x64"
+      },
+      {
+        "target": "deb",
+        "arch": "x64"
+      },
+      {
+        "target": "rpm",
+        "arch": "x64"
+      }
+    ]
+  },
+  "mac": {
+    "icon": "resources/app-icons/atom-community.png",
+    "category": "Development"
+  },
+  "win": {
+    "icon": "resources/app-icons/atom-community.ico",
+    "target": [
+      { "target": "nsis" },
+      { "target": "portable" }
+    ]
+  },
+  "extraMetadata": {
+  }
+}
+
+async function main() {
+  const package = await fs.readFile('package.json', "utf-8")
+  options.extraMetadata = generateMetadata(JSON.parse(package))
+  builder.build({
+    config: options
+  }).then((result) => {
+    console.log("Built binaries")
+    fs.mkdir('binaries').catch(() => "")
+    Promise.all(result.map(r => fs.copyFile(r, path.join('binaries', path.basename(r)))))
```

Comment:
All the promises should be handled.

```suggestion
    await Promise.all(result.map(r => fs.copyFile(r, path.join('binaries', path.basename(r)))))
```